### DISCUSSION TITLE
Why does Fish use things such as `fopen` and `fprintf` instead of streams

### DISCUSSION THREAD

**Discussion by lumynou5** [fish-shell/fish-shell#9820] [General]
Is there any reason of this? like the performance or functionality

Example of uses:
https://github.com/fish-shell/fish-shell/blob/3b555637695a860042840c9ba000e3a60f419c64/src/fish.cpp#L459

**Comment by faho** [comment]
Fish used to be a C program (`std::cout` and such streams are a C++ feature), and streams are quite slow.

**Reply by lumynou5** [reply]
Got it. thx
